### PR TITLE
Implement EfficientNet

### LINKED ISSUE #106: Add EfficientNet with pretrained weights
@pxl-th has implemented EfficientNet models in his repo https://github.com/pxl-th/EfficientNet.jl and also implemented the infrastructure to load the weights from pytorch's checkpoints. If the author agrees, we could copy or move the code here and possibly extend the weight loading to other models.

### PR DESCRIPTION
**Pull request by darsnack**
Follow-up from #106 (closes #106).

### CODE REVIEW

File: src/convnets/efficientnet.jl [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+- `inchannels`: number of input channels
+- `nclasses`: number of output classes
+- `max_width`: maximum number of output channels before the fully connected
+               classification blocks
+"""
+function efficientnet(scalings, block_config;
+                      inchannels = 3, nclasses = 1000, max_width = 1280)
+    wscale, dscale = scalings
+    out_channels = _round_channels(32, 8)
+    stem = Chain(Conv((3, 3), inchannels => out_channels;
```
Comment (theabhirath) [inline src/convnets/efficientnet.jl:27]: Isn't this just `conv_bn`?

File: src/convnets/efficientnet.jl [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    - `s`: kernel stride
+    - `e`: expansion ratio
+    - `i`: block input channels
+    - `o`: block output channels (will be scaled by global width scaling)
+- `inchannels`: number of input channels
+- `nclasses`: number of output classes
+- `max_width`: maximum number of output channels before the fully connected
+               classification blocks
+"""
+function EfficientNet(scalings, block_config;
```
Comment (theabhirath) [inline src/convnets/efficientnet.jl:119]: Isn't exported yet, I reckon

**Comment by theabhirath** [comment]
Is there also a way we can port the weights functionality from #113? Would be a good complement to #164

**Comment by darsnack** [comment]
Probably using the same technique as #164. I think any model that is plain convolutions, batch norm, and fully connected can be done in this way. There a gist linked on the HF repo model cards if you want to tackle adding this for other models in the repo.

**Comment by darsnack** [comment]
Test errors are happening later for the same memory issues as always. I've already limited the tests to the smallest subset that seems necessary.

**Comment by theabhirath** [comment]
Seems all good, but I'm worried that EfficientNets, which are the least heavy of the models, are causing this....or maybe it's a cumulative effect from the other models (Is `GC.gc()` perfect?)

**Review by theabhirath — APPROVED** [review]

**Comment by darsnack** [comment]
The tests get passed EfficientNet and error around ConvNeXt. Maybe ConvNeXt is right on the threshold for the CI?

**Comment by theabhirath** [comment]
Seems like it was the MLPMixers after all, nice!

**Comment by theabhirath** [comment]
Could maybe try not testing for the `ConvNeXt` and `ConvMixer` large models, but I'm not sure if those are the only pain points for memory here - `Xception` and `InceptionResNetv2` were pretty bad as well
